### PR TITLE
BLAS level 2

### PR DESCRIPTION
**Pull request by ben-albrecht**
This PR adds BLAS 2 support to the Chapel BLAS module.

**Progress**

- Implementation
    - [X] BLAS 2 routines
    - [X] Additional routine assertions, e.g. square, symmetric, Hermitian matrices...
- Tests
    - [X] Simple tests (1 per routine)
    - [X] Full testing coverage (op, uplo, diag, lda, ..)
    - [X] Dense (general) matrix routines
    - [X] Symmetric matrix routines
    - [X] Hermitian matrix routines
    - [X] Triangular matrix routines
- Documentation
    - [X] Simple documentation (name, equations)
    - [X] MKL URLs instead of netlib (for all BLAS routines)
- Testing
    - [X] Passes OS X + netlib BLAS testing
    - [X] Passes Cray + CrayBLAS testing

**Slated for a future PR**

- Tests
    - Band matrix routines
    - Packed matrix routines
    - Cleaner error messages replacing where-clauses
    - Tests for error-cases (halts/compiler-errors)

### CODE REVIEW

File: test/modules/packages/blas/blas_helpers.chpl [hunk 1]
```diff
@@ -54,8 +57,43 @@ proc makeHerm(A : [?Adom]) {
...
+        A[i,j] = (0):t;
+      }
+    }
+  }
+}
+
+// Make band-diagonal triangular matrix
+proc makeBandTriangular(A:[?Adom], k,  uplo:Uplo=Uplo.Upper)
+  where Adom.rank == 2 {
+
```
Comment (benharsh) [inline test/modules/packages/blas/blas_helpers.chpl]: Empty function?

File: modules/packages/BLAS.chpl [hunk 2]
```diff
@@ -683,691 +718,1646 @@ module BLAS {
...
+   TODO -- Get banded array routines working
+
+  */
+  proc tbsv(A: [?Adom] ?eltType, X: [?vDom] eltType,
+            trans : Op = Op.N,
+            order : Order = Order.Row,
+            uplo : Uplo = Uplo.Upper,
+            diag : Diag = Diag.NonUnit,
+            ldA : int = 0, incx : c_int = 1)
+            where (Adom.rank == 2) && (vDom.rank == 1)
```
Comment (benharsh) [inline modules/packages/BLAS.chpl]: I wonder if these 'where' clauses would be better as compilerError checks in the function body. In its current state someone would run into an unresolved call error, which isn't exactly all that useful.

What do you think?

File: test/modules/packages/blas/test_blas2.chpl [hunk 3]
```diff
@@ -0,0 +1,1690 @@
+use Random;
+use BLAS;
+use blas_helpers;
+
+
+proc main() {
+    // TODO -- band Matrices & packed Matrices
```
Comment (benharsh) [inline test/modules/packages/blas/test_blas2.chpl]: Can you add another TODO (unless I missed it somewhere) concerning testing for compile-time and execution-time errors?

File: test/modules/packages/blas/blas_helpers.chpl (hunk 1 above)
Comment (ben-albrecht) [inline test/modules/packages/blas/blas_helpers.chpl]: Intended as a TODO skeleton function (in conjunction with the band/packed matrix support)

File: test/modules/packages/blas/test_blas2.chpl (hunk 3 above)
Comment (ben-albrecht) [inline test/modules/packages/blas/test_blas2.chpl]: Sure, I can document this.

**Pull request merged**